NumberOfIslands.py

```diff
@@ -0,0 +1,38 @@
+# Time Complexity : O(M * N) As it traverse the whole array
```

Comment:
For dfs, in the worst case if it traverses the whole array then it would be m^2 * n^2